BUILD.bazel

```diff
@@ -0,0 +1,130 @@
+load(
+    "@gz//bazel/skylark:build_defs.bzl",
+    "GZ_FEATURES",
+    "GZ_ROOT",
+    "GZ_VISIBILITY",
+    "gz_configure_header",
+    "gz_export_header",
+    "gz_include_header",
+)
+
+package(
+    default_visibility = GZ_VISIBILITY,
+    features = GZ_FEATURES,
+)
+
+licenses(["notice"])  # Apache-2.0
+
+exports_files(["LICENSE"])
+
+gz_configure_header(
+    name = "physics_config_hh",
+    src = "include/gz/physics/config.hh.in",
+    cmakelists = ["CMakeLists.txt"],
+    defines = {
+        # These definitions are unused,
+        # this is merely to suppress generator warnings
+        "GZ_PHYSICS_ENGINE_INSTALL_DIR": "unused",
+    },
+    package = "physics",
+)
+
+gz_export_header(
+    name = "include/gz/physics/Export.hh",
+    export_base = "GZ_PHYSICS",
+    lib_name = "gz-physics",
+    visibility = ["//visibility:private"],
+)
+
+public_headers_no_gen = glob([
+    "include/gz/physics/*.hh",
+    "include/gz/physics/detail/*.hh",
+])
+
+sources = glob(
+    ["src/*.cc"],
+    exclude = [
+        "src/*_TEST.cc",
+    ],
+)
+
+gz_include_header(
+    name = "physics_hh_genrule",
```

Comment:
nit: do we want to use `physics_hh_genrule` or `physicshh_genrule`? We're using `utilshh_genrule` in https://github.com/gazebosim/gz-utils/blob/gz-utils2/BUILD.bazel